lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -207,60 +208,75 @@ ParseResult CompRegOp::parse(OpAsmParser &parser, OperationState &result) {
       return failure();
   }
 
-  SmallVector<OpAsmParser::UnresolvedOperand, 4> operands;
+  constexpr size_t clockEnabledOffset = (size_t)ClockEnabled;
+  SmallVector<OpAsmParser::UnresolvedOperand, 5> operands;
   if (parser.parseOperandList(operands))
     return failure();
   switch (operands.size()) {
   case 0:
     return parser.emitError(loc, "expected operands");
   case 1:
     return parser.emitError(loc, "expected clock operand");
-  case 2:
+  case 2 + clockEnabledOffset:
     // No reset.
     break;
-  case 3:
+  case 3 + clockEnabledOffset:
     return parser.emitError(loc, "expected resetValue operand");
-  case 4:
+  case 4 + clockEnabledOffset:
     // reset and reset value included.
     break;
   default:
+    if (ClockEnabled && operands.size() == 2)
+      return parser.emitError(loc, "expected clock enable");
     return parser.emitError(loc, "too many operands");
   }
 
   Type ty;
   if (parser.parseOptionalAttrDict(result.attributes) || parser.parseColon() ||
       parser.parseType(ty))
     return failure();
-  Type i1 = IntegerType::get(result.getContext(), 1);
 
   setNameFromResult(parser, result);
 
   result.addTypes({ty});
-  if (operands.size() == 2)
-    return parser.resolveOperands(operands, {ty, i1}, loc, result.operands);
-  else
-    return parser.resolveOperands(operands, {ty, i1, i1, ty}, loc,
-                                  result.operands);
+
+  Type i1 = IntegerType::get(result.getContext(), 1);
+  SmallVector<Type, 5> operandTypes;
+  operandTypes.append({ty, i1});
+  if (ClockEnabled)
+    operandTypes.push_back(i1);
```

Comment:
nit: `if constexpr (ClockEnabled)` should apply here

Reply:
Did not know about this!